### PR TITLE
Update the bootstrap mirror

### PR DESCRIPTION
**Pull request by alalazo**
Update the bootstrap mirror used by Spack, including:
- [x] Add support for Python 3.11
- [x] Add support for Apple M1
- [x] Update patchelf to v0.16.1
- [x] Update GnuPG to v2.3.8

### CODE REVIEW

File: lib/spack/docs/getting_started.rst [hunk 1]
```diff
@@ -178,7 +178,7 @@ Spack fall back to bootstrapping from sources:
 .. code-block:: console
 
    $ spack bootstrap disable github-actions-v0.3
-   ==> "github-actions-v0.2" is now disabled and will not be used for bootstrapping
+   ==> "github-actions-v0.3" is now disabled and will not be used for bootstrapping
```
Comment (trws) [inline lib/spack/docs/getting_started.rst:156]: Is this true? It looks enabled and still listed above.

Comment (alalazo) [reply lib/spack/docs/getting_started.rst:156]: That part is showcasing what the command:
```console
$ spack bootstrap disable github-actions-v0.3
```
does. It is unrelated to the PR, and I can submit it as a typo fix. I found it searching for `github-actions-v0.2` in the repo.

Comment (alalazo) [reply lib/spack/docs/getting_started.rst:156]: I also have to update it when this is done to add the new repo.

Comment (trws) [reply lib/spack/docs/getting_started.rst:156]: Gotcha.

**Review by haampie — APPROVED** [review]

**Pull request merged**